sources/Launcher/test/TestRobustnessAnalysisLauncher.cpp

```diff
@@ -175,5 +210,51 @@ TEST(TestLauncher, TestRobustnessAnalysisLauncher) {
   launcher.launch();
 }
 
+TEST(TestLauncher, TestRobustnessAnalysisLauncherWithCriticalTime) {
+  MyLauncher launcher;
+
+  launcher.testInputFile("");
+  launcher.setInputFile("/home/MyInputFile");
+  launcher.testInputFile("/home/MyInputFile");
+
+  launcher.testOutputFile("");
+  launcher.setOutputFile("MyOutputFile");
+  launcher.testOutputFile("MyOutputFile");
+
+  launcher.testDirectory("");
+  launcher.setDirectory("MyDirectory");
+  launcher.testDirectory("MyDirectory");
+
+  launcher.testNbThreads(1);
+  launcher.testNbThreads(1);
+
+  ASSERT_THROW_DYNAWO(launcher.init(), DYN::Error::GENERAL, DYNAlgorithms::KeyAlgorithmsError_t::DirectoryDoesNotExist);
+  launcher.testWorkingDirectory(createAbsolutePath("MyDirectory", current_path()));
+
+  launcher.setDirectory("/home/MyDirectory");
+  ASSERT_THROW_DYNAWO(launcher.init(), DYN::Error::GENERAL, DYNAlgorithms::KeyAlgorithmsError_t::DirectoryDoesNotExist);
+  launcher.testWorkingDirectory("/home/MyDirectory");
+
+  launcher.setDirectory("");
+  ASSERT_THROW_DYNAWO(launcher.init(), DYN::Error::GENERAL, DYNAlgorithms::KeyAlgorithmsError_t::FileDoesNotExist);
+  launcher.testWorkingDirectory(current_path()+"/");
+
+  launcher.setInputFile("res/MyDummyInputFile.xml");
+  ASSERT_THROW_DYNAWO(launcher.init(), DYN::Error::GENERAL, DYNAlgorithms::KeyAlgorithmsError_t::FileDoesNotExist);
+
+  launcher.setInputFile("res/MyInputFile.txt");
+  ASSERT_THROW_DYNAWO(launcher.init(), DYN::Error::GENERAL, DYNAlgorithms::KeyAlgorithmsError_t::InputFileFormatNotSupported);
+
+  boost::shared_ptr<zip::ZipFile> archive = zip::ZipFileFactory::newInstance();
+  archive->addEntry("res/CriticalTime/fic_MULTIPLE.xml");
+  zip::ZipOutputStream::write("res/CriticalTime/MyInputFile.zip", archive);
+  launcher.setInputFile("MyInputFile.zip");
+  launcher.setDirectory("res/CriticalTime");
+  launcher.setOutputFile("MyOutputFile.zip");
+  ASSERT_NO_THROW(launcher.init());
+  launcher.testOutputFileFullPath(createAbsolutePath("res/CriticalTime/MyOutputFile.zip", current_path()));
+  launcher.testMultipleJobsCriticalTime();
+  launcher.launchCriticalTime();
```

Comment:
What is the objective of this test? if it is a test for the critical time algorithm then it should be in its own file and we should test the output of the algorithm for several key scenarios

Reply:
CriticalTime inherits from Robustness, so I juts put a test in this test file. I saw that any file was created for margin calculation or systematic analysis